pkg/resources/cloudconfig/configmap.go

```diff
@@ -266,6 +297,7 @@ func getVsphereCloudConfig(
 			Datacenter:       dc.Spec.VSphere.Datacenter,
 			DefaultDatastore: datastore,
 			WorkingDir:       cluster.Name,
+			ClusterID:        dc.Spec.VSphere.Cluster,
```

Comment:
Shouldn't this be set only in the case when we're generating the CSI config? Otherwise, if we set it for cloudConfig as well, it will cause Kubelet to fail.

Reply:
The data structure is the same, and the cluster-ID is assigned in both cases (simple `cloudConfig` and `CSICloudConfig`), but they are converted to a string by using two different templates: the machine controller one (that ignores the `clusterID`) for the original `cloudConfig` and [this](https://github.com/kubermatic/kubermatic/blob/b62408de16a3f0743f050481bce033b8522bdbf7/pkg/resources/cloudconfig/vsphere/csi-configmap.go#L30) one for the CSI config, that takes into account also the `clusterID`.